src/js/popup/popup.js

```diff
@@ -53,6 +53,80 @@ function getEducationalStrings() {
   };
 }
 
+function resetNonPremiumPanel() {
+  const endOfIntroPricingElem = document.querySelector(".end-of-intro-pricing");
+  const nonPremiumPanel = document.querySelector(".content-wrapper");
+  const panelStatus = document.querySelector(".panel-status");
+
+  endOfIntroPricingElem.classList.add("is-hidden");
+  nonPremiumPanel.classList.remove("is-hidden");
+  panelStatus.classList.remove("is-hidden");
+}
+
+// End of intro pricing banner
+function showCountdownTimer() {
+  setRemainingTimeParts();
+  const timeInterval = setInterval(() => {
+   const remainingTimeInMs =  setRemainingTimeParts();
+
+  // When timer runs out, set it back to default non premium view
+   if (remainingTimeInMs <= 0) {
+    clearInterval(timeInterval);
+    resetNonPremiumPanel();
+   }
+   }, 1000);
+
+   function getRemainingTimeParts(remainingMilliseconds) {
+    const remainingDays = Math.floor(
+      remainingMilliseconds / (1000 * 60 * 60 * 24)
+    );
+    const remainingHours = Math.floor(
+      (remainingMilliseconds - remainingDays * (1000 * 60 * 60 * 24)) /
+        (1000 * 60 * 60)
+    );
+    const remainingMinutes = Math.floor(
+      (remainingMilliseconds -
+        remainingDays * (1000 * 60 * 60 * 24) -
+        remainingHours * (1000 * 60 * 60)) /
+        (1000 * 60)
+    );
+    const remainingSeconds = Math.floor(
+      (remainingMilliseconds -
+        remainingDays * (1000 * 60 * 60 * 24) -
+        remainingHours * (1000 * 60 * 60) -
+        remainingMinutes * (1000 * 60)) /
+        1000
+    );
+  
+    return {
+      remainingDays,
+      remainingHours,
+      remainingMinutes,
+      remainingSeconds,
+    }
+  }
+
+    function setRemainingTimeParts() {
+      const countdownTimer = document.querySelector(".countdown-timer");
+      const introPricingOfferEndDate = new Date(Date.UTC(2022, 8, 27, 16));
```

Comment:
To be safe, it might be better to obtain this date from the API (it's returned from `/runtime_data/` now). Just in case we need to change the date for whatever reason, it'd be great not to have to scramble to get an updated version of the add-on published.

Reply:
Actually, I see that in `background.js`, we're still getting per-country Premium availability from the deprecated `/premium_countries` API, rather than `/runtime_data`. So this could probably be relatively easily done by renaming `storePremiumAvailabilityInCountry` to `storeRuntimeData`, make it point to the `/runtime_data` endpoint, and then in addition to `premiumCountries` to also store `INTRO_PRICING_END`.

Reply:
Done, the end date and premium countries is being fetched from `/runtime_data` now.